cadence/contracts/connectors/evm/UniswapV3SwapConnectors.cdc

```diff
@@ -0,0 +1,779 @@
+            return SwapConnectors.BasicQuote(
+                inType: reverse ? self.outType() : self.inType(),
+                outType: reverse ? self.inType() : self.outType(),
+                inAmount: amountOut != nil ? safeAmountProvided : 0.0,
+                outAmount: amountOut != nil ? amountOut! : 0.0
+            )
+        }
+
+        /// Swap exact input -> min output using Uniswap V3 exactInput/Single
+        access(all) fun swap(quote: {DeFiActions.Quote}?, inVault: @{FungibleToken.Vault}): @{FungibleToken.Vault} {
+            let minOut = quote?.outAmount ?? self.quoteOut(forProvided: inVault.balance, reverse: false).outAmount
+            return <- self._swapExactIn(exactVaultIn: <-inVault, amountOutMin: minOut, reverse: false)
+        }
+
+        /// Swap back (exact input of residual -> min output)
+        access(all) fun swapBack(quote: {DeFiActions.Quote}?, residual: @{FungibleToken.Vault}): @{FungibleToken.Vault} {
+            let minOut = quote?.outAmount ?? self.quoteOut(forProvided: residual.balance, reverse: true).outAmount
+            return <- self._swapExactIn(exactVaultIn: <-residual, amountOutMin: minOut, reverse: true)
+        }
+
+        /* --- Core swap / quote internals --- */
+
+        /// Build Uniswap V3 path bytes: address(20) + fee(uint24) + address(20) + ...
+        access(self) fun _buildPathBytes(reverse: Bool): EVM.EVMBytes {
+            var bytes: [UInt8] = []
+            var i: Int = 0
+            let last = self.tokenPath.length - 1
+
+            while i < self.tokenPath.length - 1 {
+                let idx0 = reverse ? (last - i) : i
+                let idx1 = reverse ? (last - (i + 1)) : (i + 1)
+
+                let a0 = self.tokenPath[idx0]
+                let a1 = self.tokenPath[idx1]
+
+                let feeIdx = reverse ? (self.feePath.length - 1 - i) : i
+                let f: UInt32 = self.feePath[feeIdx]
+
+                // address 0
+                let a0Fixed: [UInt8; 20] = a0.bytes
+                var k: Int = 0
+                while k < 20 { bytes.append(a0Fixed[k]); k = k + 1 }
+
+                // fee uint24 big-endian
+                bytes.append(UInt8((f >> 16) & 0xFF))
+                bytes.append(UInt8((f >> 8) & 0xFF))
+                bytes.append(UInt8(f & 0xFF))
+
+                // address 1
+                let a1Fixed: [UInt8; 20] = a1.bytes
+                k = 0
+                while k < 20 { bytes.append(a1Fixed[k]); k = k + 1 }
+
+                i = i + 1
+            }
+            return EVM.EVMBytes(value: bytes)
+        }
+
+        access(self) fun to20(_ b: [UInt8]): [UInt8; 20] {
+            if b.length != 20 { panic("to20: need exactly 20 bytes") }
+            return [
+                b[0],  b[1],  b[2],  b[3],  b[4],
+                b[5],  b[6],  b[7],  b[8],  b[9],
+                b[10], b[11], b[12], b[13], b[14],
+                b[15], b[16], b[17], b[18], b[19]
+            ]
+        }
```

Comment:
Can just use `b.toConstantSized<[UInt8; 20]>()!`.

[ref](https://cadence-lang.org/docs/language/values-and-types/arrays#variable-size-array-functions)